### DISCUSSION TITLE
Physics Object spawn issues

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#8427] [Engine Core]
### Describe the project you are working on

basic 3D game with mutliplayer

### Describe the problem or limitation you are having in your project

players require positions set before being added to the scene, otherwise players will collide with each other on spawn resulting in players being stacked ontop of one and other rather than on spawn positions which of course undesireable.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

remove that dreaded error message caused by modified transforms of objects outside of the scene tree

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

allow the user to modify the global transforms before the object is added to the tree

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I can ignore the error messages but cannot toterate them.

### Is there a reason why this should be core and not an add-on in the asset library?

it only makes sense to spawn physics objects in the desired location

**Comment by ghost** [comment]
This issue feels like a form of psychological torture, as if the Engine is demanding I harm myself.

For context I am running a dedicated server and it spawns and despawns players freqently resulting in a lot of error messages and not the yellow ones but the red ones.

**Comment by AThousandShips** [comment]
Please do not vent your frustration here, and do not joke about self harm please, see the [code of conduct](https://godotengine.org/code-of-conduct/)

**Comment by Mickeon** [comment]
Have you tried modifying their `position` instead? That can be done without error, I do recall.

**Reply by ghost** [reply]
isn't that a 2D exclusive feature?

**Reply by AThousandShips** [reply]
How do you mean? That property exists in 3D as well

**Comment by Calinou** [comment]
This is being implemented in https://github.com/godotengine/godot/pull/70443 (which still needs work before it can be merged).

**Reply by ghost** [reply]
Thanks
